mongo/single_result.go

```diff
@@ -89,11 +93,6 @@ func (sr *SingleResult) setRdrContents() error {
 	case sr.cur != nil:
 		defer sr.cur.Close(context.TODO())
 
-		// Set contents of rdr to cur.Current if non-nil. Otherwise, iterate underlying
-		// cursor.
-		if sr.rdr = sr.cur.Current; sr.rdr != nil {
-			return nil
-		}
```

Comment:
To check, was this condition redundant with L103-104, or was this a necessary behavior change?

Reply:
Good catch. I added this snippet because I thought the RDR contents were not being updated correctly with `Cursor`s made with `NewCursorFromDocuments`. The code actually works as-is, and I just needed to correctly set the underlying batch of the `Cursor` [here](https://github.com/mongodb/mongo-go-driver/pull/806/files#diff-55ebb68f028f8f8bde882a1bf09524690d2b5fd42b8a04ccaa8969d39f5483b8R103-R106), so this change is no longer needed.